packages/rum-recorder/src/boot/recorder.ts

```diff
@@ -31,11 +31,27 @@ export function startRecording(
 
   lifeCycle.subscribe(LifeCycleEventType.SESSION_RENEWED, takeFullSnapshot)
   lifeCycle.subscribe(LifeCycleEventType.VIEW_CREATED, takeFullSnapshot)
+  const { stop: stopTrackingFocusRecords } = trackFocusRecords(lifeCycle, addRawRecord)
 
   return {
     stop() {
       stopRecording()
       stopSegmentCollection()
+      stopTrackingFocusRecords()
     },
   }
 }
+
+export function trackFocusRecords(lifeCycle: LifeCycle, addRawRecord: (record: RawRecord) => void) {
+  function addFocusRecord() {
+    addRawRecord({
+      type: RecordType.Focus,
+      data: {
+        has_focus: document.hasFocus(),
```

Comment:
What was the reasoning to choose focus rather than visibility?

Reply:
When multiple browser windows are open, multiple replays can be visible, while a single replay is focused. This is closer of what we want.

Reply:
The only thing I can think of is: if we detect that the user has two (or more) visible window, we might show them side by side (even though the only one will has focus and show stuff). You can always add `is_visible` flag later one if we decide this is the behaviour we want to track